src/Network/Response.php

```diff
@@ -519,7 +520,10 @@ public function sendHeaders()
         $codeMessage = $this->_statusCodes[$this->_status];
         $this->_setCookies();
         $this->_sendHeader("{$this->_protocol} {$this->_status} {$codeMessage}");
-        $this->_setContentType();
+
+        if (in_array($this->_status, [304, 204])) {
+            $this->_setContentType();
+        }
```

Comment:
This does not seem equivalent to what we had before. The intention was to avoid setting a content type on those 2 cases, but now it is somehow the opposite.